### ISSUE TITLE
TypeError: array_search(): Argument #2 ($haystack) must be of type array, null given...

### ISSUE THREAD

**Issue by pbchase** [ctsit/linear_data_entry_workflow#58]
In the [REDCap Community](https://redcap.vanderbilt.edu/community/post.php?id=206967), [mapule.nhlapho](https://redcap.vanderbilt.edu/community/profile.php?id=412) reports:

> [mapule.nhlapho](https://redcap.vanderbilt.edu/community/profile.php?id=412) asked • 6 days, 30 minutes ago (2023-05-24 09:43)
> Linear Data workflow External Module
> Hi [@philip.chase](https://redcap.vanderbilt.edu/community/profile.php?id=426) , [@taryn.stoffs](https://redcap.vanderbilt.edu/community/profile.php?id=504) , [@kyle.chesney](https://redcap.vanderbilt.edu/community/profile.php?id=6503) 
>
> REDCap 13.4.13 · PHP 8.1.2 (Linux/Unix OS) · MySQL 8.0.33
> 
> We are constantly receiving the following after enabling the EM in four projects. How can we fix this?

```
The 'linear_data_entry_workflow' module threw the following exception when calling the hook method 'hook_every_page_top':

TypeError: array_search(): Argument #2 ($haystack) must be of type array, null given in /var/www/html-orig/redcap/modules/linear_data_entry_workflow_v2.6.0/ExternalModule.php:203
Stack trace:
#0 /var/www/html-orig/redcap/modules/linear_data_entry_workflow_v2.6.0/ExternalModule.php(203): array_search()
#1 /var/www/html-orig/redcap/modules/linear_data_entry_workflow_v2.6.0/ExternalModule.php(43): LinearDataEntryWorkflow\ExternalModule\ExternalModule->loadRFIO()
#2 /var/www/html-orig/redcap/redcap_v13.4.13/ExternalModules/classes/ExternalModules.php(3162): LinearDataEntryWorkflow\ExternalModule\ExternalModule->hook_every_page_top()
#3 /var/www/html-orig/redcap/redcap_v13.4.13/ExternalModules/classes/ExternalModules.php(3327): ExternalModules\ExternalModules::startHook()
#4 /var/www/html-orig/redcap/redcap_v13.4.13/ExternalModules/classes/ExternalModules.php(3360): ExternalModules\ExternalModules::ExternalModules\{closure}()
#5 /var/www/html-orig/redcap/redcap_v13.4.13/Classes/Hooks.php(42): ExternalModules\ExternalModules::callHook()
#6 /var/www/html-orig/redcap/redcap_v13.4.13/Classes/HtmlPage.php(144): Hooks::call()
#7 /var/www/html-orig/redcap/redcap_v13.4.13/Views/HeaderProject.php(42): HtmlPage->PrintHeader()
#8 /var/www/html-orig/redcap/redcap_v13.4.13/ProjectGeneral/header.php(4): require('/var/www/html-o...')
#9 /var/www/html-orig/redcap/redcap_v13.4.13/DataEntry/record_home.php(4): include('/var/www/html-o...')
#10 {main}
```

**Comment by ChemiKyle** [comment]
Error shows up in an email when accessing Record Status Dashboard. Error is caused by this line:

https://github.com/ctsit/linear_data_entry_workflow/blob/732347aa932be7ff68bf856958516dadaeb026a6/ExternalModule.php#L203

On the Record Status Dashboard page, `$instrument` and `$event_id` are both `null`.

```php
var_dump(array_search(null, null));
```

In PHP 7.4.33 this returns null along with a warning, however, in PHP 8.1.19 this line throws an error.

**Comment by pbchase** [comment]
Addressed by PR #59
